gaas-ui/src/components/login/login-options.tsx

```diff
@@ -0,0 +1,34 @@
+import { Button, Grid } from "@material-ui/core";
+import React from "react";
+
+const sanitizeUrl = require("@braintree/sanitize-url").sanitizeUrl;
+
+interface IProps {
+    cognitoLoginURL: string;
+}
+export default function LoginOptions(props: IProps) {
+    const { cognitoLoginURL } = props;
+    const sanitizer = (url: string): string => {
+        const regex = new RegExp("[^-A-Za-z0-9+&@#/%?=~_|!:,.;()]");
+        if (regex.test(url)) {
+            return "";
+        }
+        return sanitizeUrl(cognitoLoginURL);
+    };
+    return (
+        <main id="login-options">
+            <Grid container direction="column" justify="center" alignItems="center">
+                <Button // nosemgrep
+                    id="login-with-cognito-button"
+                    aria-label="login-with-cognito-button"
+                    variant="contained"
+                    color="primary"
+                    style={{ marginTop: "20px" }}
+                    href={sanitizer(cognitoLoginURL)} // nosemgrep
```

Comment:
I presume this is safe despite semgrep warning due to the santizer?

Reply:
Yes the semgrep warning was that the URL needed to be sanitised. However, we played around with a bunch of different sanitisers and it didn't seem to get rid of the warning. I assume this is because it cannot pick up what the "sanitizer" function is doing.